packages/host/app/components/operator-mode/submode-layout.gts

```diff
@@ -139,9 +140,13 @@ export default class SubmodeLayout extends Component<Signature> {
     this.args.onSearchSheetOpened?.();
   }
 
-  @action private handleCardSelectFromSearch(card: CardDef) {
-    this.args.onCardSelectFromSearch(card);
-    this.closeSearchSheet();
+  @action private async handleCardSelectFromSearch(cardId: string) {
+    let cardResource = getCard(this, () => cardId);
+    await cardResource.loaded;
```

Comment:
This pattern is all over our codebase and it's bad. We shouldn't make, discard, and leak a cardResource. We should have a function `(cardId: string) => Promise<CardDef | undefined>`. I could have sworn we had such a method on cardService before, but it's not there anymore.

Reply:
I've (re?)implemented it and used it.